service/controller/chart/v1/resource/status/create.go

```diff
@@ -0,0 +1,66 @@
+package status
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/helmclient"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/chart-operator/service/controller/chart/v1/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCustomResource(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	releaseName := key.ReleaseName(cr)
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("getting status for release %#q", releaseName))
+
+	releaseContent, err := r.helmClient.GetReleaseContent(ctx, releaseName)
+	if helmclient.IsReleaseNotFound(err) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("release %#q not found", releaseName))
+
+		// Return early. We will retry on the next execution.
+		return nil
+	} else if err != nil {
+		return microerror.Mask(err)
+	}
+
+	releaseHistory, err := r.helmClient.GetReleaseHistory(ctx, releaseName)
+	if helmclient.IsReleaseNotFound(err) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("release %#q not found", releaseName))
+
+		// Return early. We will retry on the next execution.
+		return nil
+	} else if err != nil {
+		return microerror.Mask(err)
+	}
+
+	currentStatus := v1alpha1.ChartStatus{
+		AppVersion:   releaseHistory.AppVersion,
+		Status:       releaseContent.Status,
+		LastDeployed: v1alpha1.DeepCopyTime{releaseHistory.LastDeployed},
```

Comment:
This is the deep copy magic we need for time.Time fields.

Reply:
Ordering would be grand.